torch/distributions/distribution.py

```diff
@@ -60,3 +65,37 @@ def enumerate_support(self):
             Variable or Tensor iterating over dimension 0.
         """
         raise NotImplementedError
+
+    def _extended_shape(self, sample_shape=()):
+        """
+        Returns the size of the sample returned by the distribution, given
+        a `sample_shape`. Note, that the batch and event shapes of a distribution
+        instance are fixed at the time of construction. If this is empty, the
+        returned shape is upcast to (1,).
+
+        Args:
+            sample_shape (torch.Size): the size of the sample to be drawn.
+        """
+        shape = sample_shape + self._batch_shape + self._event_shape
+        if not shape:
+            shape = (1,)
+        return shape
+
+    def _validate_log_prob_arg(self, value):
+        """
+        Argument validation for `log_prob` methods. The rightmost dimensions
+        of a value to be scored via `log_prob` must agree with the distribution's
+        batch and event shapes.
+
+        Args:
+            value (Tensor or Variable): the tensor whose log probability is to be
+                computed by the `log_prob` method.
+        Raises
+            ValueError: when the rightmost dimensions of `value` do not match the
+                distribution's batch and event shapes.
+        """
+        if not (torch.is_tensor(value) or isinstance(value, Variable)):
```

Comment:
This doesn't take event_shape into account (which is fine for now I guess, but it would be worth mentioning somewhere)

Reply:
Actually, we can include it right now itself. I had refactored to move this into distribution, so it should handle the generic case.